### PR TITLE
str #19338: NPE in ActorGraphInterpreter (fwd port)

### PR DESCRIPTION
**Pull request by drewhk**
(cherry picked from commit 161e01e)

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala [hunk 1]
```diff
@@ -269,12 +269,12 @@ private[stream] object ActorGraphInterpreter {
       }
 
     def exposedPublisher(publisher: ActorPublisher[Any]): Unit = {
+      exposedPublisher = publisher
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: Is there any risk for overwriting? (if exposePublisher is called twice?)

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: Not really.

Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: Shouldn't we guard against that invariant then?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: Why? There are thousands of such lines where you can check the validity of a previous value (null here) and if I add an assert here, then why not add assertions there? I.e. where do you draw the line?

Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: Should this method do anything if it is invoked twice?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: It simply shouldn't be called twice.

Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: And how does a reader of this code know that?

Comment (rkuhn) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: We only send one ExposedPublisher message, which is documented accordingly. Endre is right that this is not useful to express in code—just like in many other places.

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: I can add a comment if that helps.

Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala]: Comment would be nice! :)

**Pull request merged**